compiler/ztests/summarize-lhs-error.yaml

```
@@ -0,0 +1,9 @@
+script: |
+  ! zc -s 'count() by this[a] := key'
+  ! zc -s 'this[a] := count() by key'
+
```

Add coverage for the OpAssignment case.
```suggestion
  ! zc -s 'this[a] := count()'

```